executor/adapter.go

```diff
@@ -892,14 +893,24 @@ func (a *ExecStmt) SummaryStmt() {
 	planGenerator := func() string {
 		return plannercore.EncodePlan(a.Plan)
 	}
-	_, planDigest := getPlanDigest(a.Ctx, a.Plan)
+	// Generating plan digest is slow, only generate it once if it's 'Point_Get'.
+	var planDigest string
+	var planDigestGen func() string
+	if a.Plan.TP() == plancodec.TypePointGet {
+		planDigestGen = func() string {
+			_, planDigest := getPlanDigest(a.Ctx, a.Plan)
+			return planDigest
+		}
+	} else {
+		_, planDigest = getPlanDigest(a.Ctx, a.Plan)
+	}
 
 	execDetail := stmtCtx.GetExecDetails()
 	copTaskInfo := stmtCtx.CopTasksDetails()
 	memMax := stmtCtx.MemTracker.MaxConsumed()
 	var userString string
 	if sessVars.User != nil {
-		userString = sessVars.User.String()
+		userString = sessVars.User.Username
```

Comment:
Attention, `User.Username` doesn't contain the `User.Hostname`, are you sure about this change?

If `User.String()` is slow, how bout optimize the funtion, we can use ` string + string...` instead of `fmt.Sprintf`.

Reply:
I'm not sure whether host name helps when analyzing performance. According to your experience, is it helpful?

Reply:
I rarely use `use.host` info.